apps/go-mod-validator/src/github.ts

```diff
@@ -23,35 +23,62 @@ async function isCommitInDefaultBranch(
     head: commitSha,
   });
 
-  return status === "identical" || status === "behind";
+  const isInDefault = status === "identical" || status === "behind";
+  return {
+    isInDefault,
+    commitSha,
+  };
 }
 
 async function isTagInDefaultBranch(
   gh: Octokit,
   defaultBranch: string,
   mod: GoModuleWithTag,
-): Promise<boolean> {
-  const { data: repoTags } = await gh.rest.repos.listTags({
-    owner: mod.owner,
-    repo: mod.repo,
-  });
+): Promise<{
+  isInDefault: boolean;
+  commitSha: string;
+}> {
+  let commitSha = "";
+  try {
+    const tag = await gh.rest.git.getRef({
+      owner: mod.owner,
+      repo: mod.repo,
+      ref: `tags/${mod.tag}`,
+    });
 
-  for (const repoTag of repoTags) {
-    if (repoTag.name != mod.tag) {
-      continue;
+    // lightweight tags show up as "commit" type, annotated tags show up as "tag" type
+    const isAnnotatedTag = tag.data.object.type === "tag";
+    if (isAnnotatedTag) {
+      const getTagResp = await gh.rest.git.getTag({
+        owner: mod.owner,
+        repo: mod.repo,
+        tag_sha: tag.data.object.sha,
+      });
+      commitSha = getTagResp.data.object.sha;
+    } else {
+      commitSha = tag.data.object.sha;
     }
 
     return isCommitInDefaultBranch(gh, defaultBranch, {
       ...mod,
-      commitSha: repoTag.commit.sha,
+      commitSha,
     });
+  } catch (e) {
+    core.error(`Error checking tag in default branch: ${e}`);
```

Comment:
Include the owner, repo, and sha?

Reply:
Good catch

Reply:
I made it such that there's a warning annotation that includes this context now